src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -56,9 +68,18 @@ public class PatientEverythingService : IPatientEverythingService
             string continuationToken,
             CancellationToken cancellationToken)
         {
-            EverythingOperationContinuationToken token = string.IsNullOrEmpty(continuationToken)
-                ? new EverythingOperationContinuationToken(0, null)
-                : EverythingOperationContinuationToken.FromString(ContinuationTokenConverter.Decode(continuationToken));
+            EverythingOperationContinuationToken token;
+
+            try
+            {
+                token = string.IsNullOrEmpty(continuationToken)
+                    ? new EverythingOperationContinuationToken(0, null)
+                    : EverythingOperationContinuationToken.FromString(ContinuationTokenConverter.Decode(continuationToken));
+            }
+            catch (JsonReaderException)
```

Comment:
JsonSerializationException might be thrown here as well. If the request is "$everything?ct=ew==", the decoded token will be "{" and JsonSerializationException will be thrown when deserializing it.

Reply:
Maybe we can catch `JsonException` instead.

Reply:
Good catch, thanks! Fixed in 4cf5fbd (and moved the logic to the `EverythingOperationContinuationToken` class).